tests/unit/test_cd.py

```diff
@@ -0,0 +1,18 @@
+import os
+
+from pyfakefs import fake_filesystem_unittest
+
+from needy.cd import cd, current_directory
+
+
+class CDTest(fake_filesystem_unittest.TestCase):
+    def setUp(self):
+        self.setUpPyfakefs()
+
+    def test_cd(self):
+        root = current_directory()
+        self.fs.CreateDirectory(os.path.join('tmp', 'a', 'b'))
+        with cd(os.path.join('tmp', 'a')) as a:
+            self.assertEqual(current_directory(), os.path.join(root, 'tmp', 'a'))
+            with cd('b') as b:
+                self.assertEqual(current_directory(), os.path.join(root, 'tmp', 'a', 'b'))
```

Comment:
Kind of both. There was an issue with using the manager with relative paths. If you cd into a relative path, then cd into something else, when you exited the inner cd, it would attempt to go back to where it was using the relative path and break. See the change to needy/cd.py. Without that, `-C` with a relative path breaks horribly.